### DISCUSSION TITLE
Sessions/Multi user on the WebSocket implementation for BB2

### DISCUSSION THREAD

**Discussion by renziver** [facebookresearch/ParlAI#4057] [Q&A]
How could we implement a session-like feature for the websocket service implementation? I've tested the Web Browser demo and it is not possible for a single running server of BB2 to be used by multiple users at the same time. 

if user 1 is talking with the bot, and user 2 tried to talk to the bot as well, the conversation between user 1 and the bot is carried over to the bot's dialogue with user 2.

I have read somewhere that BB2 Web Browser is currently implemented to do a one-to-one chat, how would it be possible to implement a one-to-many approach?

**Comment by klshuster** [comment]
Which web browser demo have you tried? The current implementation of ParlAI chat services should provide one-to-many functionality. One way to achieve this is using the Facebook Messenger integration; any service in general using the Websocket implementation should also handle this.

**Reply by renziver** [reply]
Hi @klshuster I've tried the browser demo from the ParlAI repository. I have opted to develop my own interface using the WebSocket implementation instead and it indeed allows me to have a one-to-many functionality.
